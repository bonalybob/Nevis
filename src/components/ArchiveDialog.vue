<template>
  <div
    class="fixed top-0 pt-8 flex w-full h-full bg-blue-point4 flex-col justify-around content-center select-none"
  >
    <div
      class="-mt-16 relative bg-white justify-around content-center text-center flex-none shadow-md mx-12 sm:mx-16 md:mx-20 px-3 pt-2 pb-3 select-none"
    >
      <h1>Multiple Records Found</h1>
      <p v-if="listOfRecords.length > 5" class="select-none">
        More Than 5 Records Were Found - Only the First 5 Are Displayed. Please
        Try Narrowing Your Criteria
      </p>
      <table class="w-full font-body mt-1">
        <tr class="font-heading text-center hover:bg-blue-light">
          <th>Name</th>
          <th>SIID</th>
          <th>Club</th>
          <th>Age Class</th>
        </tr>
        <tr
          v-for="record of trimmedListOfRecords"
          :key="record.id"
          @click="confirmAction(record)"
          class="text-center even:bg-blue-lightest hover:bg-blue-light"
        >
          <td>{{ record.name }}</td>
          <td>{{ record.siid }}</td>
          <td>{{ record.club }}</td>
          <td>{{ calculateAgeClass(record.gender, record.yearOfBirth) }}</td>
        </tr>
      </table>

      <button
        @click="confirmAction(false)"
        class="bg-white text-blue border border-blue font-heading px-4 py-1 mr-2 mt-3 select-none hover:bg-blue-point2"
      >
        Cancel
      </button>
    </div>
  </div>
</template>

<script>
import ageClassFunctions from '@/scripts/ageClass'

export default {
  name: 'ConfirmationDialog',

  props: {
    listOfRecords: {
      type: Array,
      default: () => [],
    },
  },

  computed: {
    trimmedListOfRecords: function() {
      if (this.listOfRecords.length > 5) return this.listOfRecords.slice(0, 5)
      else return this.listOfRecords
    },
  },

  methods: {
    confirmAction: function(value) {
      this.$emit('close', value)
    },
    calculateAgeClass: (gender, ageClass) =>
      ageClassFunctions(gender, ageClass),
  },
}
</script>
