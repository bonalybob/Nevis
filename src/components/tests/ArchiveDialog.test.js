import { mount } from '@vue/test-utils'
import ArchiveDialog from '@/components/ArchiveDialog'

test('Is a Vue Instance', () => {
  const wrapper = mount(ArchiveDialog, {
    stubs: ['router-link'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(ArchiveDialog, {
    stubs: ['router-link'],
  })
  wrapper.setProps({
    listOfRecords: [
      { name: 'A', siid: '1', id: 0, gender: 'm', yearOfBirth: '2000' },
      { name: 'B', siid: '2', id: 33, gender: 'f', yearOfBirth: '2001' },
      { name: 'C', siid: '3', id: 21, gender: 'm', yearOfBirth: '1989' },
    ],
  })
  expect(wrapper.element).toMatchSnapshot()
  wrapper.setProps({
    listOfRecords: [
      { name: 'A', siid: '1', id: 0, gender: 'm', yearOfBirth: '2000' },
      { name: 'B', siid: '2', id: 33, gender: 'f', yearOfBirth: '2001' },
      { name: 'C', siid: '3', id: 21, gender: 'm', yearOfBirth: '1989' },
      { name: 'A', siid: '1', id: 2, gender: 'm', yearOfBirth: '2000' },
      { name: 'B', siid: '2', id: 3, gender: 'f', yearOfBirth: '2001' },
      { name: 'C', siid: '3', id: 4, gender: 'm', yearOfBirth: '1989' },
    ],
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Emits Value on Close', () => {
  const wrapper = mount(ArchiveDialog, {
    stubs: ['router-link'],
  })
  wrapper.vm.confirmAction('Hello')
  wrapper.vm.confirmAction(123)
  expect(wrapper.emitted().close.length).toBe(2)
  expect(wrapper.emitted().close[0]).toEqual(['Hello'])
  expect(wrapper.emitted().close[1]).toEqual([123])
})

