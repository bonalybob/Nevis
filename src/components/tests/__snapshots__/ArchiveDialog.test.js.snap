// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="fixed top-0 pt-8 flex w-full h-full bg-blue-point4 flex-col justify-around content-center select-none"
>
  <div
    class="-mt-16 relative bg-white justify-around content-center text-center flex-none shadow-md mx-12 sm:mx-16 md:mx-20 px-3 pt-2 pb-3 select-none"
  >
    <h1>
      Multiple Records Found
    </h1>
     
    <!---->
     
    <table
      class="w-full font-body mt-1"
    >
      <tr
        class="font-heading text-center hover:bg-blue-light"
      >
        <th>
          Name
        </th>
         
        <th>
          SIID
        </th>
         
        <th>
          Club
        </th>
         
        <th>
          Age Class
        </th>
      </tr>
       
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          A
        </td>
         
        <td>
          1
        </td>
         
        <td>
          
        </td>
         
        <td>
          M20
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          B
        </td>
         
        <td>
          2
        </td>
         
        <td>
          
        </td>
         
        <td>
          W18
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          C
        </td>
         
        <td>
          3
        </td>
         
        <td>
          
        </td>
         
        <td>
          M21
        </td>
      </tr>
    </table>
     
    <button
      class="bg-white text-blue border border-blue font-heading px-4 py-1 mr-2 mt-3 select-none hover:bg-blue-point2"
    >
      
      Cancel
    
    </button>
  </div>
</div>
`;

exports[`Renders Correctly 2`] = `
<div
  class="fixed top-0 pt-8 flex w-full h-full bg-blue-point4 flex-col justify-around content-center select-none"
>
  <div
    class="-mt-16 relative bg-white justify-around content-center text-center flex-none shadow-md mx-12 sm:mx-16 md:mx-20 px-3 pt-2 pb-3 select-none"
  >
    <h1>
      Multiple Records Found
    </h1>
     
    <p
      class="select-none"
    >
      
      More Than 5 Records Were Found - Only the First 5 Are Displayed. Please Try Narrowing Your Criteria
    
    </p>
     
    <table
      class="w-full font-body mt-1"
    >
      <tr
        class="font-heading text-center hover:bg-blue-light"
      >
        <th>
          Name
        </th>
         
        <th>
          SIID
        </th>
         
        <th>
          Club
        </th>
         
        <th>
          Age Class
        </th>
      </tr>
       
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          A
        </td>
         
        <td>
          1
        </td>
         
        <td>
          
        </td>
         
        <td>
          M20
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          B
        </td>
         
        <td>
          2
        </td>
         
        <td>
          
        </td>
         
        <td>
          W18
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          C
        </td>
         
        <td>
          3
        </td>
         
        <td>
          
        </td>
         
        <td>
          M21
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          A
        </td>
         
        <td>
          1
        </td>
         
        <td>
          
        </td>
         
        <td>
          M20
        </td>
      </tr>
      <tr
        class="text-center even:bg-blue-lightest hover:bg-blue-light"
      >
        <td>
          B
        </td>
         
        <td>
          2
        </td>
         
        <td>
          
        </td>
         
        <td>
          W18
        </td>
      </tr>
    </table>
     
    <button
      class="bg-white text-blue border border-blue font-heading px-4 py-1 mr-2 mt-3 select-none hover:bg-blue-point2"
    >
      
      Cancel
    
    </button>
  </div>
</div>
`;
